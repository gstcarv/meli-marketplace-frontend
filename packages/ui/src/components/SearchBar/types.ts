import { ComponentProps, RefObject } from 'react';

export type SearchBarProps = ComponentProps<'input'> & {
    inputRef?: RefObject<HTMLInputElement>;
};
