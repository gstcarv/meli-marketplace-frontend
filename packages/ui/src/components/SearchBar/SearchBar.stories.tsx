import type { Meta, StoryObj } from '@storybook/react';
import { PageContainer } from '../PageContainer/PageContainer';
import { SearchBar } from './SearchBar';

const meta = {
    title: 'forms/SearchBar',
    component: SearchBar,
    parameters: {
        layout: 'centered'
    },
    tags: ['autodocs']
} satisfies Meta<typeof PageContainer>;

export const Primary: StoryObj<typeof meta> = {
   
};


export default meta;
