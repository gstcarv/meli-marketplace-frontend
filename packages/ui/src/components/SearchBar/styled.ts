import styled from 'styled-components';

export const SearchBarBase = styled.div`
    height: 40px;
    width: 40%;
    background-color: #ffffff;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
    border-radius: ${({ theme }) => theme.radii.get('xs1')};
    overflow: hidden;
    display: flex;

    .SearchBar__input {
        width: 100%;
        height: 100%;
        border: none;
        padding: 0 ${({ theme }) => theme.spacing.get('md')};
        font-size: ${({ theme }) => theme.fontSize.get('body')};
        font-weight: ${({ theme }) => theme.fontWeight.get('thin')};

        &::placeholder {
            color: ${({ theme }) => theme.colors.get('gray-600')};
        }
    }
`;

export const SearchBarButton = styled.button`
    color: ${({ theme }) => theme.colors.get('gray-600')};
    background-color: ${({ theme }) => theme.colors.get('gray-100')};
    width: 46px;
    border: none;
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;

    &:hover {
        background-color: ${({ theme }) => theme.colors.get('gray-300')};
    }
`;
