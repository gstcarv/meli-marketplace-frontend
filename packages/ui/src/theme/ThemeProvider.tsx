import 'react-loading-skeleton/dist/skeleton.css';
import { PropsWithChildren } from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { GlobalStyle } from '../globalStyles';
import { theme } from './theme';

export const ThemeProvider = (props: PropsWithChildren) => (
    <StyledThemeProvider theme={theme}>
        <>
            {props.children}

            <GlobalStyle />
        </>
    </StyledThemeProvider>
);
