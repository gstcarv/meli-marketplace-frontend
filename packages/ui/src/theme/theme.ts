import { colors, fontSize, radii, spacing, fontWeight } from '../tokens';

export const theme = { colors, fontSize, radii, spacing, fontWeight };

export type Theme = typeof theme;
