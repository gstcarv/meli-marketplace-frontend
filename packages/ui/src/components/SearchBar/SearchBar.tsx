import { SearchBarBase, SearchBarButton } from './styled';
import { SearchBarProps } from './types';
import { MagnifierIcon } from '@meli/icons';

export const SearchBar = ({ className, ...props }: SearchBarProps) => {
    return (
        <SearchBarBase className={className}>
            <input className='SearchBar__input' placeholder='Nunca deixe de buscar' {...props} />

            <SearchBarButton type='submit' aria-label='Pesquisar'>
                <MagnifierIcon size={18} />
            </SearchBarButton>
        </SearchBarBase>
    );
};
