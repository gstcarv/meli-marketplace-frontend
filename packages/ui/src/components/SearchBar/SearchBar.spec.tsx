import { fireEvent, render, screen } from '@testing-library/react';
import { ThemeProvider } from '../../theme';
import { SearchBar } from './SearchBar';

describe('<SearchBar />', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ThemeProvider>
                <SearchBar />
            </ThemeProvider>
        );

        expect(container).toMatchSnapshot();
    });

    it('should call onChange when any change happens on input text', () => {
        const onChangeFn = jest.fn();

        render(
            <ThemeProvider>
                <SearchBar placeholder='mock-placeholder' onChange={onChangeFn} />
            </ThemeProvider>
        );

        const searchInput = screen.getByPlaceholderText('mock-placeholder');

        fireEvent.change(searchInput, { target: { value: 'mock writing' } });

        expect(onChangeFn).toHaveBeenCalledTimes(1);
    });

    it('should submit form when search button clicked', () => {
        const onSubmitFn = jest.fn((e) => e.preventDefault());

        render(
            <ThemeProvider>
                <form onSubmit={onSubmitFn}>
                    <SearchBar placeholder='mock-placeholder' onChange={onSubmitFn} />
                </form>
            </ThemeProvider>
        );

        const searchButton = screen.getByRole('button', { name: 'Pesquisar' });

        fireEvent.click(searchButton);

        expect(onSubmitFn).toHaveBeenCalledTimes(1);
    });
});
