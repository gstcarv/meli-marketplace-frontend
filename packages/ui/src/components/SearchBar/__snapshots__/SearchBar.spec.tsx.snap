// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> should render correctly 1`] = `
<div>
  <div
    class="sc-blHHSb hboDbF"
  >
    <input
      class="SearchBar__input"
      placeholder="Nunca deixe de buscar"
    />
    <button
      aria-label="Pesquisar"
      class="sc-gtLWhw dXamHe"
      type="submit"
    >
      <svg
        class="icon-magnifier"
        fill="none"
        height="18"
        size="18"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
          stroke="none"
        />
        <path
          d="M10 10m-7 0a7 7 0 1 0 14 0a7 7 0 1 0 -14 0"
        />
        <path
          d="M21 21l-6 -6"
        />
      </svg>
    </button>
  </div>
</div>
`;
